spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/InstanceDetailsProviderMock.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.spark.metrics.sink.statsd;
+
+import java.util.Optional;
+
+public class InstanceDetailsProviderMock extends InstanceDetailsProvider {
```

Comment:
any reasons for not using Mockito instead?

Reply:
Now that I used PowerMock for InstanceDetailsProvider tests, there is no reason not to use Mockito here as well. Changed as suggested.